proposals/0015-partitioned-epoch-reward-distribution.md

```diff
@@ -0,0 +1,362 @@
+
+   - rewards distribution: distribute the epoch rewards for the active stake
+     accounts
+
+## Alternatives Considered
+
+We have discussed three alternative approaches.
+
+- The first approach is to set a threshold on stake balance, and limit the epoch
+rewards to the accounts with stake balance above the threshold. This will
+effectively reduce the number of stake rewards to be distributed, and reduce
+reward distribution time at the epoch boundary. However, this will impact stake
+accounts with small balance. To receive rewards, the small stake accounts will
+be forced to join stake pools, which some may be hesitant to do.
+
+- The second approach is to handle reward distributions through transactions with
+specialized native reward programs. While this approach is optimal, it requires
+significant modifications and doesn't align with the current reward code.
+
+- The third approach is a completely asynchronous epoch rewards calculation and
+distribution, in which both reward computation and rewards distribution are
+asynchronous. This is a more general approach. However, it is also a more
+complex approach. The reward computation states would have to be maintained
+across multiple blocks. The transition between the reward calculation and
+reward distribution would need to be synchronized in the cluster. And cluster
+restart during reward computation would have to be handled specially.
+
+## Detailed Design
+
+The major bottleneck for epoch reward distribution is to distribute rewards to
+stake accounts. At the time of writing, there are approximately 550K active
+stake accounts and 1.5K vote accounts on Solana Mainnet Beta. Given the
+relatively small number of vote accounts, it makes sense to keep vote rewards
+distribution mechanism unchanged. They can still be distributed efficiently at
+the first block of the epoch boundary. This reduces the impact of rewards for
+vote account and also simplifies the overall changes. It also lets us focus on
+solving the primary bottleneck - Stake Rewards. Only Stake rewards are going to
+be distributed out over multiple blocks.
+
+In the new stake rewards distribution approach, we will separate the
+computation of rewards from the actual distribution of rewards at the epoch
+boundary by dividing the process into two distinct phases:
+
+   1. rewards calculation phase - during which the epoch rewards for all
+      activate stake accounts are computed and distribution chunks are scheduled.
+
+   1. rewards distribution phase - during which the calculated epoch rewards
+      for the active stake accounts are distributed.
+
+To help track and verify the reward distribution in the next phase, two new
+kinds of SysVar accounts are proposed : (a) `EpochRewardHistory` SysVar
+account and (b) `EpochRewardReserve` SysVar accounts. These SysVar accounts are
+updated at the boundary blocks (i.e. first and last blocks) in the reward
+calculation phase. More details of these two kinds of SysVar accounts are
```

Comment:
Aren't the sysvars updated during the first block of the epoch, and also every block of the reward distribution phase?

Reply:
Only one sysvar will exist in this SIMD. 
And yes, that one sysvar will be updated during the first block of the epoch, and also every block of the reward distribution phase.